framer/Layer.coffee

```diff
@@ -1017,17 +985,18 @@ class exports.Layer extends BaseClass
 		enumerable: false
 		exportable: false
 		importable: false
-		get: -> @_stateMachine.states
+		get: ->
+			@_states ?= new LayerStates(@)
+			return @_states
 		set: (states) ->
-			@_stateMachine.reset()
-			for name, state of states
-				@_stateMachine.states[name] = state
+			@states.machine.reset()
+			_.extend(@states, states)
```

Comment:
We could capture the current state of the layer here, we should however be aware of the setting of states individually...